src/client/caching/structures/Attachment.ts

```diff
@@ -0,0 +1,94 @@
+import { APIMessageAttachmentData } from '@klasa/dapi-types';
+import { Stream } from 'stream';
+
+/**
+ * @see https://discord.com/developers/docs/resources/channel#attachment-object
+ */
+export class Attachment implements APIMessageAttachmentData {
+
+	/**
+	 * Attachment id.
+	 * @since 0.0.1
+	 */
+	public readonly id: string;
+
+	/**
+	 * Name of file attached.
+	 * @since 0.0.1
+	 */
+	public filename: string;
+
+	/**
+	 * Size of file in bytes.
+	 * @since 0.0.1
+	 */
+	public size: number;
+
+	/**
+	 * Source url of file.
+	 * @since 0.0.1
+	 */
+	public url: string;
+
+	/**
+	 * A proxied url of file.
+	 * @since 0.0.1
+	 */
+	public proxy_url: string;
```

Comment:
Should we keep this as non-camelcase?

Reply:
@bdistin ^

Reply:
yes, it `APIMessageAttachmentData` so that it can be directly sent to the API as-is. (dual purpose like Embed). We might also include a getter for the non-camel case variant for user consumption, as that wouldn't be added in the toJSON process.